### DISCUSSION TITLE
Edit SBAT information

### DISCUSSION THREAD

**Discussion by raddirad** [ipxe/ipxe#683] [General]
Hi

when building a shim I have to place an sbat.csv into the data directory to make the buildprocess use this sbat information into the binary.
Is there any such way in iPXE?
I did build an ipxe.efi and SBAT informaton is present, but how can I edit this information?

**Comment by stappersg** [comment]
> when building a shim I have to place an sbat.csv into the data directory
> to make the buildprocess use this sbat information into the binary.

I didn't know what it is.  A websearch did bring
up https://github.com/rhboot/shim/blob/main/SBAT.md


> Is there any such way in iPXE?
> I did build an ipxe.efi and SBAT informaton is present,
> but how can I edit this information?

Somehow I do read that as

* I couldn't find  sbat.csv  integration in iPXE
* I made something
* I'm NOT gonna share with you how I did that
* I don't want to rebuild it again and again
* I want you to help me

**Reply by raddirad** [reply]
I didn't make anything related to SBAT. I did build ipxe and checked for existing SBAT information with ```objdump```. So the Information is present somewhere in the code but I couldd't find a way to edit it.

```
/tftpboot/linux/pxelinux.cfg/ipxe.efi:     file format pei-x86-64

Contents of section .sbat:
 1a5000 73626174 2c312c53 42415420 56657273  sbat,1,SBAT Vers
 1a5010 696f6e2c 73626174 2c312c68 74747073  ion,sbat,1,https
 1a5020 3a2f2f67 69746875 622e636f 6d2f7268  ://github.com/rh
 1a5030 626f6f74 2f736869 6d2f626c 6f622f6d  boot/shim/blob/m
 1a5040 61696e2f 53424154 2e6d640a 69707865  ain/SBAT.md.ipxe
 1a5050 2c312c69 5058452c 69707865 2e656669  ,1,iPXE,ipxe.efi
 1a5060 2c312e32 312e312b 20286764 33633839  ,1.21.1+ (gd3c89
 1a5070 292c6874 7470733a 2f2f6970 78652e6f  ),https://ipxe.o
 1a5080 72670a                               rg.          
 ```

**Comment by NiKiZe** [comment]
There has been commits for sbat, the commit messages has some details on how it is to be used.

**Comment by mcb30** [comment]
https://github.com/ipxe/ipxe/commit/f4f9adf618cd85d330a896e1f721f3aa78d2409d#diff-bc1cae6d71e6579fe93037da3e7a93cadb8b312689ea08be24a8714af519f1b6 is the relevant part of the relevant commit

**Reply by stappersg** [reply]
If I understand that correctly, is @raddirad looking for `config/local/branding.h` as the file to edit.   (Time will tell if I did understand correctly.)
